lib/build/script_compactor.dart

```diff
@@ -81,11 +81,11 @@ class ScriptCompactor {
 
   Future<Asset> run() {
     var crawler = new ImportCrawler(transform, primaryInput, logger);
-    return crawler.crawlImports().then((imports) {
+    return crawler.crawlImports().then((imports) async {
```

Comment:
it's not a flattening bug, it is https://github.com/dart-lang/sdk/issues/25944, which I have a CL out, waiting for one more LGTM.

`async` is a good workaround, another good one is to pass the type arg explicitly like `.then/*<Future<Asset>>*/(...)`